src/logger.ts

```diff
@@ -0,0 +1,66 @@
+import minimist from "minimist";
+import winston from "winston";
+
+export interface CustomLogger extends winston.Logger {
+    error: winston.LeveledLogMethod;
+    maker: winston.LeveledLogMethod;
+    taker: winston.LeveledLogMethod;
+    info: winston.LeveledLogMethod;
+    data: winston.LeveledLogMethod;
+    verbose: winston.LeveledLogMethod;
+}
+
+const colorizer = winston.format.colorize({
+    all: true,
+    colors: {
+        error: "red",
+        maker: "cyan",
+        taker: "yellow",
+        info: "purple",
+        data: "grey",
+        verbose: "green",
+    },
+});
+
+export function createLogger() {
+    const level = minimist(process.argv.slice(2), {
```

Comment:
I'd happier if `process.XYZ` stuff happens only on the very root of the application to avoid implicit dependencies on global variables.

Can be a follow-up ticket :)

Reply:
I tend to agree, but I couldn't figure out another way to avoid passing either the logging level or the logger around.

Reply:
I'd opt for passing around the log-level. Basically extract `process.argv.slice(2)` into a parameter would already do it :)